src/rancilio-pid.cpp

```diff
@@ -1989,6 +1989,11 @@ void setup() {
                 mqtt.setCallback(mqtt_callback);
                 checkMQTT();
             }
+
+            if(INFLUXDB_USER != "" && INFLUXDB_PASSWORD != "") {
```

Comment:
String comparisons do not work this way in C/C++. Maybe better use:
`if ((strlen(INFLUXDB_USER) > 0) && (strlen(INFLUXDB_PASSWORD) > 0))`
or maybe safer (didn't tested this code):
`if ((String(INFLUXDB_USER).length() > 0) && (String(INFLUXDB_PASSWORD).length() > 0))`

Reply:
done